quail/processes/wps_climdexInput_csv.py

```diff
@@ -0,0 +1,280 @@
+                supported_formats=[
+                    Format("text/csv", extension=".csv"),
+                ],
+            ),
+            ComplexInput(
+                "tmin_file",
+                "daily minimum temperature data file",
+                abstract="Name of file containing daily minimum temperature data.",
+                min_occurs=0,
+                max_occurs=1,
+                supported_formats=[
+                    Format("text/csv", extension=".csv"),
+                ],
+            ),
+            ComplexInput(
+                "prec_file",
+                "daily total precipitation data file",
+                abstract="Name of file containing daily total precipitation data.",
+                min_occurs=1,
+                max_occurs=1,
+                supported_formats=[
+                    Format("text/csv", extension=".csv"),
+                ],
+            ),
+            ComplexInput(
+                "tavg_file",
+                "mean temperature data file",
+                abstract="Name of file containing daily mean temperature data.",
+                min_occurs=0,
+                max_occurs=1,
+                supported_formats=[
+                    Format("text/csv", extension=".csv"),
+                ],
+            ),
+            LiteralInput(
+                "na_strings",
+                "climdexInput name",
+                abstract="Strings used for NA values; passed to read.csv",
+                default="NULL",
+                data_type="string",
+            ),
+            tmax_column,
+            tmin_column,
+            prec_column,
+            tavg_column,
+            base_range,
+            cal,
+            date_fields,
+            date_format,
+            n,
+            northern_hemisphere,
+            quantiles,
+            temp_qtiles,
+            prec_qtiles,
+            max_missing_days,
+            min_base_data_fraction_present,
+            output_file,
+            vector_name,
+            log_level,
+        ]
+
+        outputs = [ci_output]
+
+        super(ClimdexInputCSV, self).__init__(
+            self._handler,
+            identifier="climdex_input_csv",
+            title="climdexInput generator (CSV)",
+            abstract="Process for creating climdexInput object from CSV files",
+            metadata=[
+                Metadata("NetCDF processing"),
+                Metadata("Climate Data Operations"),
+                Metadata("PyWPS", "https://pywps.org/"),
+                Metadata("Birdhouse", "http://bird-house.github.io/"),
+                Metadata("PyWPS Demo", "https://pywps-demo.readthedocs.io/en/latest/"),
+            ],
+            inputs=inputs,
+            outputs=outputs,
+            store_supported=True,
+            status_supported=True,
+        )
+
+    def collect_literal_inputs(self, request):
+        return [
+            arg[0] for arg in list(collect_args(request, self.workdir).values())[-19:]
```

Comment:
Do you think there is anyway to refactor this with out the specific indexing (-19)? I noticed it's In the ClimdexInput raw, you could potentially move it to utils?